react/src/lib/components/DeckGLMap/utils/continuousLegend.ts

```diff
@@ -1,5 +1,86 @@
-import { scaleSequential, interpolateViridis, ScaleSequential } from "d3";
+import { color } from "d3-color";
+import { interpolateRgb } from "d3-interpolate";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const colorTemplate = require("../../../../demo/example-data/welllayer_continuous_template.json");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const colorTables = require("../../../../demo/example-data/color-tables.json");
 
-export function interpolatorContinuous(): ScaleSequential<string, never> {
-    return scaleSequential(interpolateViridis);
+interface colorTables {
+    name: string;
+    description: string;
+    colors: [number, number, number, number][];
+}
+
+interface propertiesObj {
+    objectName: string;
+    colorTable: string;
+    context: string;
+    colorInterpolation: string;
+}
+
+type propertiesArr = Array<propertiesObj>;
+
+interface colorTemplate {
+    name: string;
+    properties: propertiesArr;
+}
+
+export function colorsArray(
+    objectName: string
+): [number, number, number, number][] {
+    const properties = colorTemplate[0]["properties"];
+    const propertiesData = properties.filter(
```

Comment:
we can use type insensitive comparison by using toLower

Reply:
Changes as per the comment